jsonld2compactjsonldldj/jsonld2compactjsonldldj.py

```diff
@@ -15,52 +43,52 @@ def run():
     optional_arguments = parser._action_groups.pop()
 
     required_arguments = parser.add_argument_group('required arguments')
-    required_arguments.add_argument('-context', type=str, help='The JSON-LD context file', required=True)
+    required_arguments.add_argument('-context', type=str, help='The JSON-LD context file')
 
     optional_arguments.add_argument('-input', type=str, help='the input JSON-LD record array')
     optional_arguments.add_argument('-record-field', type=str, dest='record_field',
                                     help='A field that should be contained in all records, e.g., a record identifier (this field will be used to identify records)')
     optional_arguments.add_argument('-context-url', type=str, dest='context_url',
                                     help='A JSON-LD context URL that should be set to reference to the JSON-LD context (instead of inline the JSON-LD context)')
+    optional_arguments.add_argument('-mp', action="store_true",
+                                    help='Switch on to use Multiprocessing')
+
 
     parser._action_groups.append(optional_arguments)
 
     args = parser.parse_args()
 
-    context = args.context
-
-    with open(context, newline='') as contextfile:
-        jsonldcontext = json.load(contextfile)
-
-    record_field = None
-
-    if args.record_field is not None:
-        record_field = args.record_field
-
-    context_url = None
-
-    if args.context_url is not None:
-        context_url = args.context_url
-
+    if args.context_url:
+        r=requests.get(args.context_url)
+        if r.ok:
+            jsonldcontext=r.json()
+            sys.stderr.write("got context from "+args.context_url+"\n")
+    elif args.context:
+        with open(args.context, newline='') as contextfile:
+            jsonldcontext = json.load(contextfile)
+    else:
+        sys.stderr.write("got no context-file or url. aborting\n")
+        exit(-1)
+        
     if args.input is not None:
         with open(args.input, newline='') as inputfile:
             jsonldarray = json.load(inputfile)
     else:
         jsonldarray = json.load(sys.stdin)
 
-    options = {'skipExpansion': True}
-
+    #options = {'skipExpansion': True}
+    if args.mp:
+        pool = Pool(initializer=init_mp,initargs=(args.mp,jsonldcontext,args.record_field,args.context_url,))
```

Comment:
I would recommend to reduce the size of the thread pool to max. 60 or 70% of available cores.